src/stateManager/StateManager.ts

```diff
@@ -650,6 +651,129 @@ class StateManager {
         }
     }
 
+    /**
+     * Updates the state snapshot when the fork is different
+     */
+    public async updateSnapshotFork(): Promise<void> {
+        try {
+            // Get the current on-chain snapshot first
+            const currentOnChainSnapshot = StateSnapshot.from(
+                await this.stateChannelManagerContract.getStateSnapshot(
+                    this.channelId
```

Comment:
add a trelo card `[optimization] - use cached/local 'currentOnChainSnapshot' instead of alwways querying the RPC node` - this is low priority so but it at the bottom of the TODO list